src/js/administration/components/Api.tsx

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useMutation, useQueryClient } from "react-query";
+import { ExternalLink } from "../../base";
+import { updateSettings } from "../api";
+import { settingsQueryKeys } from "../querys";
+import { Settings } from "../types";
+import { SettingsCheckbox } from "./SettingsCheckbox";
+
+type ApiProps = {
+    /** The settings data */
+    data: Settings;
+};
+
+/**
+ * A component managing JSON API settings, allowing users to toggle external API access
+ */
+export default function Api({ data }: ApiProps) {
```

Comment:
```suggestion
export default function Api({ settings }: ApiProps) {
```
the name `data` is more vague than it needs to be. We know that it should just be settings so we can be more specific. Also you might be able to destructure `enable_api` out right here in the declaration.